modules/ztp-setting-managed-bare-metal-host-kernel-arguments-gitops-workflow.adoc

```diff
@@ -0,0 +1,97 @@
+
+[NOTE]
+====
+In {product-title} {product-version}, you can only add kernel arguments. You can not replace or delete kernel arguments.
+====
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (oc).
+* You have logged in to the hub cluster as a user with cluster-admin privileges.
+
+.Procedure
+
+. Create the `InfraEnv` CR and edit the `spec.kernelArguments` specification to configure kernel arguments.
+
+.. Save the following YAML in an `InfraEnv-example.yaml` file:
++
+[NOTE]
+====
+The `InfraEnv` CR has template variables such as `{{ .Cluster.ClusterName }}`. The `SiteConfig` CR automatically populates values for these templates during deployment. Do not edit the templates manually.  
+====
++
+[source,yaml,options="nowrap",role="white-space-pre"]
+----
+apiVersion: agent-install.openshift.io/v1beta1
+kind: InfraEnv
+metadata:
+  annotations:
+    argocd.argoproj.io/sync-wave: "1"
+  name: "{{ .Cluster.ClusterName }}"
+  namespace: "{{ .Cluster.ClusterName }}"
+spec:
+  clusterRef:
+    name: "{{ .Cluster.ClusterName }}"
+    namespace: "{{ .Cluster.ClusterName }}"
+  kernelArguments:
+    - operation: append <1>
+      value: audit=0 <2>
+    - operation: append 
+      value: trace=1
+  sshAuthorizedKey: "{{ .Site.SshPublicKey }}"
+  proxy: "{{ .Cluster.ProxySettings }}"
+  pullSecretRef:
+    name: "{{ .Site.PullSecretRef.Name }}"
+  ignitionConfigOverride: "{{ .Cluster.IgnitionConfigOverride }}"
+  nmStateConfigLabelSelector:
+    matchLabels:
+      nmstate-label: "{{ .Cluster.ClusterName }}"
+  additionalNTPSources: "{{ .Cluster.AdditionalNTPSources }}" 
+----
+<1> Specify the append operation to add a kernel argument.
+<2> Specify the kernel argument you want to configure. This example configures the audit kernel argument and the trace kernel argument. 
+
+. Commit the `InfraEnv-example.yaml` CR to the same location in your Git repository that has the `SiteConfig` CR and push your changes.
+
+. Edit the `spec.clusters.crTemplates` specification in the `SiteConfig` CR to reference the `InfraEnv-example.yaml` CR in your Git repository:
++
+[source,yaml,options="nowrap",role="white-space-pre"]
+----
+clusters:
+  crTemplates:
+    InfraEnv: "InfraEnv-example.yaml"
+----
++
+When you are ready to deploy your cluster by committing and pushing the `SiteConfig` CR, the build pipeline uses the custom `InfraEnv-example` CR in your Git repository to configure the infrastructure environment, including the custom kernel arguments. 
+
+.Verification
+To verify that the kernel arguments are applied to the managed bare-metal host, enter a debug session on the target host before the cluster is installed and view the kernel arguments in the `/proc/cmdline` file.
```

Comment:
```suggestion
To verify that the kernel arguments are applied to the managed bare-metal host, enter a debug session on the target host before the cluster is installed and view the kernel arguments in the `/proc/cmdline` file.
```

Is it possible to open an `oc` debug session on the host before it's fully installed?